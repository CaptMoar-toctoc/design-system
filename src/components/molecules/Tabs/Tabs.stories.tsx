import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import Tabs from './Tabs';


export default {
    title: 'TocToc-DS/Molecules',
    component: Tabs,
} as ComponentMeta<typeof Tabs>;

const Template: ComponentStory<typeof Tabs> = (args) => <Tabs {...args} />;

export const TabsTT = Template.bind({});

TabsTT.args = {
    index: 1,
    tabs: [{
        text: 'Proyectos nuevos en venta',
        handlerFn: () => { },
    },

    {
        text: 'Propiedades usadas en venta',
        handlerFn: () => { }
    },
    {
        text: 'Propiedades usadas en arriendo',
        handlerFn: () => { }
    }
    ]
};