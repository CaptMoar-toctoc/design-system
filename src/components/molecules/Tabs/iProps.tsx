export interface iProps {
  index?: number,
  value?: number,
  tabs: Array<iTabs>
}

export interface iTabs {
  text: string,
  handlerFn: Function
}
