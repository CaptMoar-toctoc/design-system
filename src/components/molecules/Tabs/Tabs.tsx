import * as React from 'react';
import { FC } from 'react';
import { Tabs, Tab, Box, ThemeProvider } from '@mui/material';
import { iProps } from './iProps';
import theme from '../../theme/theme';

const idProps = (index: number) => {
	return {
		id: `quotes-tab-${index}`,
		'aria-controls': `my-quotes-option-${index}`,
	};
}

const TabsTT: FC<iProps> = ({
	tabs,
	index = 0
}
) => {
	const [value, setValue] = React.useState(index);

	const handleChange = (event: React.SyntheticEvent, newValue: number) => {
		setValue(newValue);
	};

	return (
		<ThemeProvider theme={theme}>
			<Box sx={{ width: '100%' }}>
				<Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
					<Tabs value={value} onChange={handleChange} aria-label="my-quotes-section">
						{tabs.map((tab, index) => <Tab label={tab.text} {...idProps(index)} onClick={tab.handlerFn()} />)}
					</Tabs>
				</Box>
			</Box>
		</ThemeProvider>
	);
}

export default TabsTT;