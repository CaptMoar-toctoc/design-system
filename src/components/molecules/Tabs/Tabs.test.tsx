import Tabs from './Tabs';
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';


describe("Tabs testing", () => {
    const initProps = {
        tabs: [{
            text: 'Proyectos nuevos en venta',
            handlerFn: () => { }
        },

        {
            text: 'Propiedades usadas en venta',
            handlerFn: () => { }
        },
        {
            text: 'Propiedades usadas en arriendo',
            handlerFn: () => { }
        }
        ]
    }

    test("Render Tabs component", () => {
        render(<Tabs {...initProps} />)
    })

    test("Render SnapShot", () => {
        expect(render(<Tabs {...initProps} />)).toMatchSnapshot();
    })

    test.todo("It has a rendered text"), () => {
        render(<Tabs {...initProps} />);
        const text = screen.queryByText('Proyectos')
        expect(text).not.toBeNull();
    }

    test.todo("Is a clickable component"), () => {
        render(<Tabs {...initProps} />)
        const Tab = screen.getByRole('option')
        fireEvent.click(Tab)
        expect(Tab).toBeTruthy();
    }
})

describe('Tabs Testing renders with empty props', () => {

    const initProps = {
        tabs: [
        ]
    };

    test('Render Tabs component with empty props', () => {
        render(<Tabs {...initProps} />)
        expect(<Tabs {...initProps} />).toBeTruthy();
    })
})

export { }