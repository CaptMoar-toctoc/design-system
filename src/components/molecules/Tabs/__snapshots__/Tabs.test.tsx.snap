// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs testing Render SnapShot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-8atqhb"
      >
        <div
          class="MuiBox-root css-1gsv261"
        >
          <div
            class="MuiTabs-root css-1ujnqem-MuiTabs-root"
          >
            <div
              class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
              style="overflow: hidden; margin-bottom: 0px;"
            >
              <div
                aria-label="my-quotes-section"
                class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                role="tablist"
              >
                <button
                  aria-controls="my-quotes-option-0"
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-16msyux-MuiButtonBase-root-MuiTab-root"
                  id="quotes-tab-0"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  Proyectos nuevos en venta
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-controls="my-quotes-option-1"
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-16msyux-MuiButtonBase-root-MuiTab-root"
                  id="quotes-tab-1"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  Propiedades usadas en venta
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-controls="my-quotes-option-2"
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-16msyux-MuiButtonBase-root-MuiTab-root"
                  id="quotes-tab-2"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  Propiedades usadas en arriendo
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <span
                class="MuiTabs-indicator css-6z7m36-MuiTabs-indicator"
                style="left: 0px; width: 0px;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-8atqhb"
    >
      <div
        class="MuiBox-root css-1gsv261"
      >
        <div
          class="MuiTabs-root css-1ujnqem-MuiTabs-root"
        >
          <div
            class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
            style="overflow: hidden; margin-bottom: 0px;"
          >
            <div
              aria-label="my-quotes-section"
              class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
              role="tablist"
            >
              <button
                aria-controls="my-quotes-option-0"
                aria-selected="true"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-16msyux-MuiButtonBase-root-MuiTab-root"
                id="quotes-tab-0"
                role="tab"
                tabindex="0"
                type="button"
              >
                Proyectos nuevos en venta
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-controls="my-quotes-option-1"
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-16msyux-MuiButtonBase-root-MuiTab-root"
                id="quotes-tab-1"
                role="tab"
                tabindex="-1"
                type="button"
              >
                Propiedades usadas en venta
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-controls="my-quotes-option-2"
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-16msyux-MuiButtonBase-root-MuiTab-root"
                id="quotes-tab-2"
                role="tab"
                tabindex="-1"
                type="button"
              >
                Propiedades usadas en arriendo
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <span
              class="MuiTabs-indicator css-6z7m36-MuiTabs-indicator"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
